internal/services/synapse/synapse_workspace_resource.go

```diff
@@ -496,7 +489,12 @@ func resourceSynapseWorkspaceRead(d *pluginsdk.ResourceData, meta interface{}) e
 	d.Set("resource_group_name", id.ResourceGroup)
 	d.Set("location", location.NormalizeNilable(resp.Location))
 
-	if err := d.Set("identity", flattenIdentity(resp.Identity)); err != nil {
+	flattenIdenties, err := flattenIdentity(resp.Identity)
+	if err != nil {
+		return fmt.Errorf("retrieving Managed Identities for %s: %+v", *id, err)
```

Comment:
(same here) this isn't retrieving the identities? since these are common methods we can just return this directly:

```suggestion
		return err
```